packages/x-data-grid/src/material/index.tsx

```diff
@@ -388,11 +388,11 @@ function transformInputProps(props: GridSlotProps['baseInput'] | undefined) {
     result.endAdornment = <InputAdornment position="end">{result.endAdornment}</InputAdornment>;
   }
 
-  if (slotProps?.htmlInput) {
+  if (slotProps?.input) {
```

Comment:
This and the change in `gridBaseSlots` might be a slight breaking change, which should be flagged. 🤔

Reply:
Maybe @mui/xgrid could assert it and handle it accordingly? 🤔 
It looks like this might have been a mistake all along. 🙈

Reply:
What `slotProps` should be used for the HTML input element? Did it change?

Reply:
- In the `Input` components, `slotProps.input` points to the input HTML element
- In the `TextField` component, `slotProps.htmlInput` points to the input HTML element, and `slotProps.input` to the `Input` component

It hasn't changed

Reply:
Ok. So the `slotProps` here aren't material `slotProps`, they're data-grid `baseInput` `slotProps` which have their own independent set of [typings](https://github.com/mui/mui-x/blob/df806a70a766eafd8e539f2fa449afad78e2a745/packages/x-data-grid/src/models/gridBaseSlots.ts#L227). I didn't notice that nuance in material props when I wrote the grid base slot typings, so I made `htmlInput` refer to the HTML input element for both `baseInput` and `baseTextField`.

The grid base slots are inspired by material API but don't match it in many places. It's possible that we keep using `htmlInput` in both cases, and I think it's going to be more intuitive for grid-bindings authors reading those typings, so I'd keep it as it is. Any thoughts on that?

Reply:
Thanks for clarifying. 👍 
Yeah, it makes sense to me, in such case, the changes made by Diego should be reverted. 👌

Reply:
Ok, we have a problem if we revert this change.
Docs examples with Material components start failing prop checking, because they do not expect to receive `slotProps.htmlInput` on a InputBase slot. 🙈 
So, this also potentially means a BC for users if we go with some sort of override. 🤔 

Either we need to rename the slotProps to `input` as initially done by Diego or consider alternative typing approaches. 🤷 
WDYT @romgrk? Does the current state make sense?
`InputBase` component never had a `htmlInput` slot, it was always `input`, so, if I understand it correctly, material migrating to `slotProps` triggered this error as now the typing is very close, but different, while before there were no `slotProps`. 🤔

Reply:
> material migrating to slotProps triggered this error as now the typing is very close, but different

In this case it's close but different, but in other cases it's completely different.

> Docs examples with Material components start failing prop checking, because they do not expect to receive slotProps.htmlInput on a InputBase slot.

Why is an `InputBase` receiving those props though? The grid `baseInput` is not an `InputBase`, typecheck should not fail. Which examples are failing?

Reply:
Context: https://mui-org.slack.com/archives/C011VC970AW/p1741974665990299

So it seems like we're going to accept that `baseInput != InputBase`, and that props need to be adapted from base slots to material-ui.